finetune/adapter.py

```diff
@@ -140,8 +141,10 @@ def main(fabric: L.Fabric, devices: int, seed: int, config: Config, io: IOArgs,
         fabric.print(f"Memory used: {torch.cuda.max_memory_allocated() / 1e9:.02f} GB")
 
     # Save the final checkpoint at the end of training
-    save_path = io.out_dir / "lit_model_adapter_finetuned.pth"
+    save_path = io.out_dir / "lit_model.pth"
     save_adapter_checkpoint(fabric, model, save_path)
+    # Copy checkpoint files from original checkpoint dir
+    copy_config_files(io.checkpoint_dir, io.out_dir)
```

Comment:
I like the approach in general. Thanks for working on it. 

As proposed in #923, I suggest to make each checkpoint a folder itself. Right now you are saving to `out_dir` directly. I would make `lit_model_adapter_finetuned` a folder and save `lit_model.pth` and the other files into this. And we would also have to do it for intermediate checkpoint files, in the loop below. It makes the most sense to me as it would allow you to flexibly name your checkpoints and at the same time make them very portable.

Reply:
So concretely, I would propose:

save_path = io.out_dir / "lit_model_adapter_finetuned" / "lit_model.pth"
save_adapter_checkpoint(fabric, model, save_path)
# Copy checkpoint files from original checkpoint dir
copy_config_files(io.checkpoint_dir, save_path.parent)

Reply:
I like this idea, because in some cases it prevents accidentally overriding existing LoRA checkpoints when running an Adapter script etc. Happy to make that change.